docs/platform/connector-development/config-based/understanding-the-yaml-file/property-chunking.md

```diff
@@ -0,0 +1,309 @@
+    type:
+      type: string
+      enum: [PropertiesFromEndpoint]
+    property_field_path:
+      description: Describes the path to the field that should be extracted
+      type: array
+      items:
+        type: string
+      examples:
+        - ["name"]
+      interpolation_context:
+        - config
+        - parameters
+    retriever:
+      description: Requester component that describes how to fetch the properties to query from a remote API endpoint.
+      anyOf:
+        - "$ref": "#/definitions/SimpleRetriever"
+        - "$ref": "#/definitions/CustomRetriever"
+    $parameters:
+      type: object
+      additionalProperties: true
+```
+
+### `GroupByKeyMergeStrategy`
+
+```yaml
+GroupByKeyMergeStrategy:
+  title: Group by Key
+  description: Record merge strategy that combines records according to fields on the record.
+  required:
+    - type
+    - key
+  properties:
+    type:
+      type: string
+      enum: [GroupByKeyMergeStrategy]
+    key:
+      title: Key
+      description: The name of the field on the record whose value will be used to group properties that were retrieved through multiple API requests.
+      anyOf:
+        - type: string
+        - type: array
+          items:
+            type: string
+      examples:
+        - "id"
+        - ["parent_id", "end_date"]
+    $parameters:
+      type: object
+      additionalProperties: true
+```
+
+## Property limit types
+
+### Characters
+
+When using `characters` as the limit type, the total character count of all property names (including delimiters) determines chunk size.
+
+### Property count
+
+When using `property_count` as the limit type, the number of individual properties determines chunk size.
+
+## Record merging
+
+When the connector needs multiple requests to fetch all properties for a record, it must merge the results back together. The `GroupByKeyMergeStrategy` combines records based on a specified key field.
+
+### Simple key merging
+
+For records with a single unique identifier.
+
+```yaml title="manifest.yaml"
+record_merge_strategy:
+  type: GroupByKeyMergeStrategy
+  key: "id"
+```
+
+### Compound key merging
+
+For records requiring multiple fields to create a unique identifier.
+
+```yaml title="manifest.yaml"
+record_merge_strategy:
+  type: GroupByKeyMergeStrategy
+  key: ["parent_id", "end_date"]
+```
+
+## Always include properties
+
+Some properties must be in every chunk request, typically because they're needed for record merging or API requirements. These properties are automatically added to each chunk and don't count toward the property limit.
+
+```yaml title="manifest.yaml"
+always_include_properties:
+  - dateRange
+  - pivotValues
+```
+
+## Usage examples
+
+Here are examples of how Airbyte has used property chunking in some connectors.
+
+### HubSpot: character-based chunking
+
+HubSpot's API has a limit on the total character count of properties you can request. Here's how to configure character-based chunking.
```

Comment:
I think the more important thing to note about this usage example is that it uses dynamic property chunking. And this mean that instead of defining a static list of properties to retrieve, the list of properties to fetch are determined by hitting `https://api.hubapi.com/properties/v2/contacts/properties`

I recommend we mention that in the title and in the description. The title being: `###: HubSpot: Fetching properties from an API endpoint with character-based chunking` . Something like that